iac-extensions/arm/src/test/resources/checks/RedundantResourceDependenciesCheck/test.json

```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "apiVersion": "2019-04-01",
+      "type": "Microsoft.Network/networkInterfaces",
+      "name": "noncompliant1",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/virtualNetworks', 'exampleVNet')]",
+        "exampleDependency",
+        "[concat('Microsoft.DBforPostgreSQL/servers/', variables('serverName'))]"
```

Comment:
1. The variable `serverName` should be defined to have a complete example
2. Please also add a negative case here (only explicit dependency without any implicit reference)